docs/varnish.md

```diff
@@ -0,0 +1,90 @@
+# Tag-based cache invalidation for Varnish
+
+This is guide on how to setup Varnish in order to use effective cache invalidation. The idea behind it is that cache tags provided by Drupal 8 are used to invalidate cache (sometimes this action will be called "purge" in the following documentation). In order to achieve tag-based cache invalidation a few modules have to be installed and configured to work in combination with customized cache invalidation scripts provided for Varnish.
+
+### Requirements
+
+1. Varnish service
+2. Purge Drupal 8 module (https://www.drupal.org/project/purge)
+
+#### Setup Varnish
+
+The first step is to setup Varnish to accept commands provided by Purge module. At first, we will add the list of servers (IPs) that are allowed to do cache invalidation. That are usually your Drupal 8 servers. The reason for whitelisting Drupal 8 servers is to avoid possible DOS attacks from public IP addresses. At the beginning of the Varnish script file the following code should be added:
+```varnish
+# Whitelist of Purger servers.
+acl purgers {
```

Comment:
"purgers" is a reserved name and varnish will not start with this. Please use a different name.